cmd/lotus-fountain/main.go

```diff
@@ -193,14 +195,34 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	to, err := address.NewFromString(r.FormValue("address"))
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
-	}
-	if to == address.Undef {
-		http.Error(w, "empty address", http.StatusBadRequest)
-		return
+	var to address.Address
+
+	addressInput := r.FormValue("address")
+	if strings.HasPrefix(addressInput, "0x") {
+		ethAddress, err := ethtypes.ParseEthAddress(addressInput)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		filecoinAddress, err := ethAddress.ToFilecoinAddress()
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		to = filecoinAddress
```

Comment:
Is it ok to skip the check for empty address with checking filecoinAddress == address.Undef like when its not a 0x address? Can we make the checks for both scenarios the same and maybe refactor the code a bit to use the same error checking code?

Reply:
`ToFilecoinAddress` can return `address.Undef` but in such case it also returns `error` so it's not strictly needed because the error is checked here. The point about having the same error handling is legit, will do.

Reply:
@fridrik01 Is this the way to go in Go https://github.com/filecoin-project/lotus/pull/10784/commits/6c25f2172e41fddb775cfb911f1681ca7b2507db ? 🙂